support-workers/src/main/scala/com/gu/zuora/subscriptionBuilders/SubscribeItemBuilder.scala

```diff
@@ -1,53 +1,109 @@
 package com.gu.zuora.subscriptionBuilders
 
-import com.gu.support.workers.states.CreateZuoraSubscriptionState
-import com.gu.support.workers.{Address, GiftRecipient, User}
+import com.gu.i18n.Currency
+import com.gu.support.catalog.ProductRatePlanId
+import com.gu.support.redemptions.RedemptionCode
+import com.gu.support.redemptions.redemptions.RawRedemptionCode
+import com.gu.support.workers.{Address, GeneratedGiftCode, PaymentMethod, SalesforceContactRecord, User}
 import com.gu.support.zuora.api._
+import com.gu.zuora.subscriptionBuilders.SubscribeItemBuilder.buildContactDetails
+import org.joda.time.{DateTimeZone, LocalDate}
 
-object SubscribeItemBuilder {
+import java.util.UUID
+
+class SubscribeItemBuilder(
+  requestId: UUID,
+  user: User,
+  currency: Currency,
+) {
 
-  def build(state: CreateZuoraSubscriptionState, subscriptionData: SubscriptionData): SubscribeItem = {
-    val billingEnabled = state.paymentMethod.isLeft
-    //Documentation for this request is here: https://www.zuora.com/developer/api-reference/#operation/Action_POSTsubscribe
+  def build(
+    subscriptionData: SubscriptionData,
+    salesForceContact: SalesforceContactRecord,
+    maybePaymentMethod: Option[PaymentMethod],
+    soldToContact: Option[ContactDetails]
+  ): SubscribeItem = {
+    val billingEnabled = maybePaymentMethod.isDefined
     SubscribeItem(
-      account = buildAccount(state),
-      billToContact = buildContactDetails(state.user, None, state.user.billingAddress),
-      soldToContact = state.user.deliveryAddress map (buildContactDetails(state.user, state.giftRecipient, _, state.user.deliveryInstructions)),
```

Comment:
bill contact is not product specific, but sold to contact is, so these have been refactored as approriate